osmdroid-android/src/main/java/org/osmdroid/util/MapTileIndex.java

```diff
@@ -0,0 +1,42 @@
+package org.osmdroid.util;
+
+import org.osmdroid.tileprovider.MapTile;
+
+import microsoft.mappoint.TileSystem;
+
+/**
+ * Computes a map tile index as `long` to/from zoom/x/y
+ * Algorithm unfortunately different from SqlTileWriter.getIndex for historical reasons.
+ * This version is better, because it's easy to get zoom, X and Y back from the index.
+ * This version is limited to zooms between 0 and 29, which should be enough.
+ * @since 6.0.0
+ * @author Fabrice Fontaine
+ */
+
+public class MapTileIndex {
+
+    private static int mMaxZoomLevel = TileSystem.primaryKeyMaxZoomLevel;
+    private static int mModulo = 1 << mMaxZoomLevel;
+
+    public static long getTileIndex(final MapTile pMapTile) {
+        return getTileIndex(pMapTile.getZoomLevel(), pMapTile.getX(), pMapTile.getY());
+    }
+
+    public static long getTileIndex(final int pZoom, final int pX, final int pY) {
+        return (((long)pZoom) << (mMaxZoomLevel * 2))
```

Comment:
i kind of feel that we should put this algorithm somewhere. it's used in several locations throughout the code base

Reply:
The algorithm is unfortunately slightly different in `SqlTileWriter.getIndex`. And mine is better ;)
If possible, we should reuse `MapTileIndex` now.